New package Zabbix.Agent.2 version 7.2.3

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #221404

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.9 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.9.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

## Additional information

> [!WARNING]
> 
> - A `x86` install wizard CANNOT be installed in a `x64` environment.
> - The install wizard CANNOT run interactively properly in a `x64` _Windows Sandbox_ instance, and immediately fail with exit code `1603`.
>   - But it CAN launch interactively properly in a `x64` Windows host device.
>   - There might be missing environment required by the wizard.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/223483)

Dragon1573 committed:
New package Zabbix.Agent.2 version 7.2.3

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

stephengillie commented:
Automatic Validation ended with:
>  Installation failed with exit code -1978335226 2025-02-05 11:22:24.191 [CLI ] ShellExecute installer failed: 1603 Property(S): ErrorDialog = ErrorDlg Property(S): SecureCustomProperties = ADK_ERROR;ALLOWDENYKEY;ARPNOREPAIR;CONF;ENABLEPATH;ENABLEPERSISTENTBUFFER;ENABLEREMOTECOMMANDS;HOSTINTERFACE;HOSTMETADATA;HOSTMETADATAITEM;HOSTNAME;INCLUDE;INCLUDE_ERROR;LISTENIP;LISTENPORT;LOGFILE;LOGTYPE;NETFRAMEWORK20;NETFRAMEWORK30;NETFRAMEWORK35;NETFRAMEWORK40FULL;NONMSICLEAR;NONMSICONFNAME;NONMSIDIRNAME;NONMSIPROGNAME;NONMSIPROGSTATE;NONMSISERVCMD;OAC;OLDUSERSID;PERSISTENTBUFFERFILE;PERSISTENTBUFFERPERIOD;READ_CONF;ROOTDRIVE;SERVER;SERVERACTIVE;SKIP;STARTUPTYPE;STATUSPORT;TIMEOUT;TLSACCEPT;TLSCAFILE;TLSCERTFILE;TLSCONNECT;TLSCRLFILE;TLSKEYFILE;TLSPSKFILE;TLSPSKFILE_ERROR;TLSPSKIDENTITY;TLSPSKVALUE;TLSPSKVALUE_NOTCHECK;TLSPSKVALUE_PROCESSED;TLSSERVERCERTISSUER;TLSSERVERCERTSUBJECT;WIX_DOWNGRADE_DETECTED;WIX_IS_NETFRAMEWORK_40_OR_LATER_INSTALLED;WIX_UPGRADE_DETECTED MSI (s) (5C:F4) [11:22:24:062]: Product: Zabbix Agent 2 (64-bit) -- Installation failed. 2025-02-05 11:27:02.680 [CLI ] ShellExecute installer failed: 1603 MSI (s) (78:F4) [11:27:02:554]: Product: Zabbix Agent 2 (32-bit) -- Installation failed.

Automated error analysis suggests -1978335226 may mean Running ShellExecute failed

Automated error analysis suggests 1603 may mean Previous MSI installer is missing (Or it might mean Install successful but system restart may be required.) Should this be add as an `InstallerSuccessCode` or similar?

(Automated response - build 977.)

Dragon1573 commented:
> Automated error analysis suggests 1603 may mean Previous MSI installer is missing (Or it might mean Install successful but system restart may be required.) Should this be add as an `InstallerSuccessCode` or similar?

NO. Error code `1603` will also pop up if you install it interactively. Disk modifications are reverted and the package is not installed on the device. It can't be configured as an `InstallerSuccessCode`.

stephengillie commented:
Manual Validation ended with:
> 11:04:07 AM: Product: Zabbix Agent 2 (64-bit) -- Please enter the name or IP address in the Zabbix server IP/DNS field or set SERVER variable on command line !

(Automated response - build 977.)

Dragon1573 commented:
> Manual Validation ended with:
> 
> ```text
> 11:04:07 AM: Product: Zabbix Agent 2 (64-bit) -- Please enter the name or IP address in the Zabbix server IP/DNS field or set SERVER variable on command line !
> ```
> 
> (Automated response - build 977.)

Emmm ... Seems these configurations can't be pre-defined in the manifest. They should be manually configured when a user install it. These configurations may various between each user.

I would prefer reject this PR and mark it <https://github.com/microsoft/winget-pkgs/labels/Blocking-Issue> <https://github.com/microsoft/winget-pkgs/labels/Interactive-Only-Installer>

Dragon1573 committed:
Update manifest: Zabbix.Agent.2 version 7.2.3

- Pass a default required version via "InstallerSwitches"
- Add acceptable "InstallerSuccessCodes"

Above changes are studied from Chocolatey Package Manager

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

Dragon1573 commented:
@AWBbox, now the <https://github.com/microsoft/winget-pkgs/labels/Interactive-Only-Installer> should be resolved 🎉 

But it seems there's still a problem running validation in a VM instance (_Windows Sandbox_ or _Microsoft Azure Cloud VM_). This may cause <https://github.com/microsoft/winget-pkgs/labels/Validation-Shell-Execute>.

Dragon1573 commented:
> My concern is that if Zabbix Agent 2 is already installed and configured with a specified server, that using the server switch of "127.0.0.1" via Winget may overwrite the value in the configuration.

I think it won't.

This package is released as an MSI install wizard. These kinds of wizard will always write Windows Registry and add themselves into the _Add & Remove Programs_ panel. WinGet can extract these "ARP entries" and detect that there's already an `Zabbix.Agent.2` instance installed.

> I am trying to confirm this. What command is Winget actually running here?

This should be explained by [moderators](https://github.com/microsoft/winget-pkgs/issues/15674). To my knowledge, it's also launched by passing those `InstallerSwitches` to `msiexec.exe`, unless you enabled the `directMSI` feature by invoking `winget settings` and manually edit the JSON configurations.

Dragon1573 commented:
> IRunning msiexec with the switches I see [here](https://github.com/microsoft/winget-pkgs/pull/223483/files#diff-d99281daa6e156ec25af4869fbd3972daa7253ba2f5e5e09af63a2d2fa36597b) does not work in a Windows command prompt. I.e.
> 
> `msiexec /qb /norestart SERVER="127.0.0.1" zabbix_agent2-7.2.2-windows-amd64-openssl.msi`

Did you try running this command in _Windows PowerShell_ or _PowerShell Core 7_ session? Did you try swapping the command as `msiexec zabbix.msi /qb ...`? I can't sure if command-line parameters' order matters.

AWBbox commented:
> > IRunning msiexec with the switches I see [here](https://github.com/microsoft/winget-pkgs/pull/223483/files#diff-d99281daa6e156ec25af4869fbd3972daa7253ba2f5e5e09af63a2d2fa36597b) does not work in a Windows command prompt. I.e.
> > `msiexec /qb /norestart SERVER="127.0.0.1" zabbix_agent2-7.2.2-windows-amd64-openssl.msi`
> 
> Did you try running this command in _Windows PowerShell_ or _PowerShell Core 7_ session? Did you try swapping the command as `msiexec zabbix.msi /qb ...`? I can't sure if command-line parameters' order matters.

![Screenshot 2025-02-08 172510](https://github.com/user-attachments/assets/e35f9483-d3dc-4647-876e-7f7b3aeb9104)

msiexec just brings a popup up when I run the command, and the behaviour is the same in Powershell and from an elevated prompt.

Dragon1573 commented:
What about adding a `/i` before `zabbix.msi`?

AWBbox commented:
That's what I was missing, thanks.

In the following example, installing 7.2.2 works fine and sets the server value in "C:\Program Files\Zabbix Agent 2\zabbix_agent2.conf" to "foo". Unfortunately, updating to 7.2.3 using the same method and different server value overwrites the value in the configuration file. This would not be good in a production environment! If an existing value exists in the configuration file then it should not be overwritten by Winget.

![screenshot1](https://github.com/user-attachments/assets/6147492f-59b3-408f-bcd3-2e8d2d87f417)

Also, the agent installs a service and the update appears to need user interaction to close when msiexec runs:

![screenshot2](https://github.com/user-attachments/assets/b491417c-c255-42be-ba8e-c9a1082a04c9)

Dragon1573 commented:
> In the following example, installing 7.2.2 works fine and sets the server value in "C:\Program Files\Zabbix Agent 2\zabbix_agent2.conf" to "foo". Unfortunately, updating to 7.2.3 using the same method and different server value overwrites the value in the configuration file. This would not be good in a production environment! If an existing value exists in the configuration file then it should not be overwritten by Winget.

Is it acceptable leaving "SERVER" property unset while performing an upgrade installation via `msiexec.exe`?

> Also, the agent installs a service and the update appears to need user interaction to close when `msiexec` runs:

This is intended. _WinGet_ will not close or terminate any running application while upgrading or uninstalling, to avoid data lost. "Terminate before upgrade" action should be done by the install wizard itself.

AWBbox commented:
The challenge is that if Winget is installing Zabbix Agent 2 for the first time, then a placeholder value for SERVER should be used. E.g. in Linux, this is "127.0.0.1", allowing the user to change the value after installation to suit their environment:

`msiexec /qb /norestart SERVER="127.0.0.1" /i zabbix_agent2-7.2.2-windows-amd64-openssl.msi`

However, if Winget is upgrading an existing installation, it should NOT specify a SERVER value in order for the configuration file to retain that setting, else it will be overwritten:

`msiexec /qb /norestart /i zabbix_agent2-7.2.3-windows-amd64-openssl.msi`

Dragon1573 committed:
Update manifest: Zabbix.Agent.2 version 7.2.3

- Add "Upgrade" property for the package

stephengillie commented:
Automatic Validation ended with:
>  Executable C:\Program Files\Zabbix Agent 2\zabbix_agent2.exe returned exit code: 1 Executable C:\Program Files\Zabbix Agent 2\zabbix_get.exe returned exit code: 1 Executable C:\Program Files\Zabbix Agent 2\zabbix_sender.exe returned exit code: 1

(Automated response - build 978.)
